diff -up readline-6.0/shlib/Makefile.in.shlib readline-6.0/shlib/Makefile.in
--- readline-6.0/shlib/Makefile.in.shlib	2009-01-06 18:03:22.000000000 +0100
+++ readline-6.0/shlib/Makefile.in	2009-07-02 14:36:15.000000000 +0200
@@ -169,7 +169,7 @@ $(SHARED_READLINE):	$(SHARED_OBJ)
 
 $(SHARED_HISTORY):	$(SHARED_HISTOBJ) xmalloc.so
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so
 
 # Since tilde.c is shared between readline and bash, make sure we compile 
 # it with the right flags when it's built as part of readline
diff -up readline-6.0/support/shlib-install.shlib readline-6.0/support/shlib-install
--- readline-6.0/support/shlib-install.shlib	2008-07-20 01:16:05.000000000 +0200
+++ readline-6.0/support/shlib-install	2009-07-02 14:37:59.000000000 +0200
@@ -71,7 +71,7 @@ fi
 case "$host_os" in
 hpux*|darwin*|macosx*|linux*)
 	if [ -z "$uninstall" ]; then
-		chmod 555 ${INSTALLDIR}/${LIBNAME}
+		chmod 755 ${INSTALLDIR}/${LIBNAME}
 	fi ;;
 cygwin*|mingw*)
 	IMPLIBNAME=`echo ${LIBNAME} \
diff -up readline-6.0/support/shobj-conf.shlib readline-6.0/support/shobj-conf
--- readline-6.0/support/shobj-conf.shlib	2009-01-04 20:32:42.000000000 +0100
+++ readline-6.0/support/shobj-conf	2009-07-02 14:40:06.000000000 +0200
@@ -112,10 +112,11 @@ sunos5*|solaris2*)
 linux*-*|gnu*-*|k*bsd*-gnu-*)
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
-	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
+ 	SHOBJ_LDFLAGS='$(CFLAGS) -shared -Wl,-soname,$@'
 
-	SHLIB_XLDFLAGS='-Wl,-rpath,$(libdir) -Wl,-soname,`basename $@ $(SHLIB_MINOR)`'
+	SHLIB_XLDFLAGS='-Wl,-soname,`basename $@ $(SHLIB_MINOR)`'
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)$(SHLIB_MINOR)'
+ 	SHLIB_LIBS='-ltinfo'
 	;;
 
 freebsd2*)
